### ISSUE TITLE
Integrate quantization pass with CompiledModule

### ISSUE THREAD

**Issue by IanNod** [nod-ai/SHARK-Turbine#183]
Currently we are calling MMGroupQuantRewriterPass in stateless_llama.py. This limits how/what we can compile quantized models to (doesn't work lowering to linalg IR). Integrating MMGroupQuantRewriterPass within the CompiledModule should fix this and make more generalizable in general.

**Comment by vivekkhandelwal1** [comment]
Hi @IanNod, do we still need this issue?

**Comment by IanNod** [comment]
> Hi @IanNod, do we still need this issue?

Yes, this has been low priority so hasn't been addressed but we should still fix this at some point
